File: 125-ValidPalindrome/125-ValidPalindrome.java
// Last updated: 8/1/2025, 10:39:46 PM
class Solution {
    public boolean isPalindrome(String s) {
        StringBuilder sb=new StringBuilder();
        for(int i=0;i<s.length();i++){
            char c=s.charAt(i);
            if((c >= 'a' && c <= 'z') || (c >= 'A' && c <= 'Z') || (c >= '0' && c <= '9')) { 
             sb.append(Character.toLowerCase(c));
            }
        }
        int i=0;
        int j=sb.length()-1;
        while(i<=j){
            if(sb.charAt(i)!=sb.charAt(j)){
                return false;
            }
            i++;
            j--;
        }
        return true;
    }
}

File: 125-ValidPalindrome/125-ValidPalindrome.py
# Last updated: 8/1/2025, 10:39:49 PM
class Solution:
    def isPalindrome(self, s: str) -> bool:
        s=s.lower()
        s="".join(c.lower() for c in s if c.isalnum())
        if s==s[::-1]:
            return True
        else:
            return False    
